src/DataTransferObject.php

```diff
@@ -318,6 +321,40 @@ public function getDefinedPropertyNames(): array
         return array_keys($this->properties);
     }
 
+    /**
+     * @param string|array $propertyNames
+     *
+     * @return void
+     */
+    public function assertDefined($propertyNames): void
+    {
+        $this->assertKnownPropertyNames($propertyNames);
+
+        $undefined = array_filter((array) $propertyNames, function (string $propertyName) {
+            return !$this->isDefined($propertyName);
+        });
+
+        if (!empty($undefined)) {
+            throw new UninitialisedPropertiesError($undefined, static::class);
+        }
+    }
+
+    /**
+     * @param string|array $propertyNames
+     *
+     * @return void
+     */
+    private function assertKnownPropertyNames($propertyNames): void
+    {
+        $unknown = array_filter((array) $propertyNames, function (string $propertyName) {
+            return !array_key_exists($propertyName, $this->propertyTypes);
+        });
```

Comment:
I wonder if an array_diff would be more performant here:

`$unknown = array_diff($propertyNames, array_keys($this->propertyTypes));`

Reply:
Yeah switching this one to array diff.